src/main/java/mate/academy/service/calculation/BalanceCalculator.java

```diff
@@ -0,0 +1,14 @@
+package mate.academy.service.calculation;
+
+import mate.academy.dao.ShopDao;
+import mate.academy.dao.ShopDaoImpl;
+import mate.academy.model.FruitTransaction;
+
+public class BalanceCalculator implements TransactionCalculation {
+    private final ShopDao shopDao = new ShopDaoImpl();
+
```

Comment:
```suggestion
public class BalanceCalculator implements TransactionCalculation {
    private final ShopDao shopDao;
    
    BalanceCalculator(ShopDao shopDao) {
    	this.shopDao = shopDao;
    }
```

It's better to initialize ShopDao through constructor